keymap/sublime.js

```diff
@@ -420,6 +420,34 @@
 
   map[cK + ctrl + "Backspace"] = "delLineLeft";
 
+  cmds[map["Backspace"] = "smartBackspace"] = function(cm) {
+    if (cm.somethingSelected()) return CodeMirror.Pass;
+
+    cm.operation(function() {
+      var cursors = cm.listSelections();
+      var indentUnit = cm.getOption("indentUnit");
+
+      for (var i = 0; i < cursors.length; i++) {
```

Comment:
As far as I can see, this will only work if you iterate over the selections in inverse order -- right now, your deletes will, if multiple cursors are on the same line, misalign later deletes, which are still under the impression that the cursor is in the old place.